application/shared-kernel/ApiCore/SinglePageApp/SinglePageAppConfiguration.cs

```diff
@@ -63,36 +64,48 @@ public string GetHtmlTemplate()
             return _htmlTemplate;
         }
         
-        var retryCount = 0;
-        while (!File.Exists(HtmlTemplatePath) && retryCount++ < 10)
-        {
-            // When running locally, this code might be called while index.html is recreated, give it a few seconds to finish.
-            Thread.Sleep(TimeSpan.FromSeconds(1));
-        }
+        AwaitSinglePageAppGeneration();
         
-        if (!File.Exists(HtmlTemplatePath))
+        if (!File.Exists(_htmlTemplatePath))
         {
-            throw new FileNotFoundException("index.html does not exist.", HtmlTemplatePath);
+            throw new FileNotFoundException("index.html does not exist.", _htmlTemplatePath);
         }
         
-        _htmlTemplate = File.ReadAllText(HtmlTemplatePath, new UTF8Encoding());
+        _htmlTemplate = File.ReadAllText(_htmlTemplatePath, new UTF8Encoding());
         return _htmlTemplate;
     }
     
+    [Conditional("DEBUG")]
+    private void AwaitSinglePageAppGeneration()
+    {
+        var startNew = Stopwatch.StartNew();
+        while (startNew.Elapsed < TimeSpan.FromSeconds(10))

```

Comment:
Let's bump it to 30, it some times takes longer

```suggestion
        while (startNew.Elapsed < TimeSpan.FromSeconds(30))
```

Reply:
10 seconds is already 5x what we need. This represents the delta between starting the .NET API and building the frontend build. If it takes longer than a few seconds, it’s likely because of something else. In case it fails, one can just refresh the browser and it will work.